test/unit/assorted/server_discovery_and_monitoring.spec.test.ts

```diff
@@ -374,15 +374,6 @@ function assertOutcomeExpectations(topology, events, outcome) {
     }
 
     expect(description).to.include.keys(key);
-
-    if (outcomeValue == null) {
-      expect(description[key]).to.be.undefined;
-    } else {
-      if (typeof outcomeValue === 'object') {
-        expect(description).to.have.property(key).that.deep.equals(outcomeValue);
-      } else {
-        expect(description).to.have.property(key, outcomeValue);
-      }
-    }
+    expect(description).to.have.deep.property(key, outcomeValue == null ? undefined : outcomeValue);
```

Comment:
Just to double check, if the outcomeValue is null, we still expect the property to exist on the description, just with a value of "undefined"?

Reply:
I've updated the code here, the remaining keys being checked where all always defined except for logicalSessionTimeoutMinutes, which is always defined but is set explicitly to undefined.